src/API/Google/AdsCampaignLabel.php

```diff
@@ -0,0 +1,162 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Query\AdsCampaignLabelQuery;
+use Automattic\WooCommerce\GoogleListingsAndAds\Google\Ads\GoogleAdsClient;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsAwareInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsAwareTrait;
+use Google\Ads\GoogleAds\Util\V16\ResourceNames;
+use Google\Ads\GoogleAds\V16\Resources\Label;
+use Google\Ads\GoogleAds\V16\Resources\CampaignLabel;
+use Google\Ads\GoogleAds\V16\Services\LabelOperation;
+use Google\Ads\GoogleAds\V16\Services\CampaignLabelOperation;
+use Google\Ads\GoogleAds\V16\Services\MutateOperation;
+use Google\Ads\GoogleAds\V16\Services\MutateGoogleAdsRequest;
+
+/**
+ * Class AdsCampaignLabel
+ * https://developers.google.com/google-ads/api/docs/reporting/labels
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Google
+ */
+class AdsCampaignLabel implements OptionsAwareInterface {
+
+	use OptionsAwareTrait;
+
+	/**
+	 * Temporary ID to use within a batch job.
+	 * A negative number which is unique for all the created resources.
+	 *
+	 * @var int
+	 */
+	protected const TEMPORARY_ID = -1;
+
+	/**
+	 * The Google Ads Client.
+	 *
+	 * @var GoogleAdsClient
+	 */
+	protected $client;
+
+
+	/**
+	 * AdsCampaignLabel constructor.
+	 *
+	 * @param GoogleAdsClient $client
+	 */
+	public function __construct( GoogleAdsClient $client ) {
+		$this->client = $client;
+	}
+
+	/**
+	 * Get the label ID by name.
+	 *
+	 * @param string $name The label name.
+	 *
+	 * @return null|int The label ID.
+	 *
+	 * @throws ApiException If the search call fails.
+	 */
+	protected function get_label_id_by_name( string $name ) {
+		$query = new AdsCampaignLabelQuery();
+		$query->set_client( $this->client, $this->options->get_ads_id() );
+		$query->where( 'label.name', $name, '=' );
+		$label_results = $query->get_results();
+		$label_id      = null;
+
+		foreach ( $label_results->iterateAllElements() as $row ) {
+			return $row->getLabel()->getId();
+		}
+
+		return $label_id;
```

Comment:
I'm curious why you decided to create the `$label_id` variable instead of just returning `null` after the loop?

Reply:
Yep that's true, it is unnecessary, I removed it here [d30851d](https://github.com/woocommerce/google-listings-and-ads/pull/2485/commits/d30851d08749e68bdbab6ecfa508d66c7c95a735)